### PR TITLE
Ripul handoo/relationships/pod template

### PR DESCRIPTION
**Pull request by RipulHandoo**
**Notes for Reviewers**

This PR Adds
1. Relationship between Pod Template and other Kubernetes resources.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by RipulHandoo** [comment]
Working Proof of this PR.

https://github.com/meshery/meshery/assets/107461226/73ce3ecf-381d-4043-9c8a-7fc2dfe8aee5

**Comment by RipulHandoo** [comment]
**_Note:_** Spreadsheet Updated. ✅

**Comment by leecalcote** [comment]
@MUzairS15 what does something like this PR mean in terms of our ability to visually represent both a Deployment component with a PodTemplate component inside of it?

Better yet, what does this mean for the user who started designing with a Pod (PodTemplate), then decided to rollup to a Deployment?

**Review by MUzairS15 — APPROVED** [review]
Will you suggest a new `subType` as discussed on the meeting, were in you can convert/upgrade a Pod to Deployment/Job/...

With your current relationship, one can duplicate PodTemplate and assign it to other resources eg: Deployment, Job, Statefulset.. removing the need to configure the object again and again. It is an improvement, thanks.

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 1]
```diff
@@ -921,6 +921,95 @@
...
+              "description": "Jobs use pod templates to specify the container images, commands, and resources required for executing batch processing tasks reliably within Kubernetes clusters, ensuring consistency and repeatability in job execution."
+            }
+          }, 
+          {
+            "kind": "CronJob",
+            "model": "kubernetes",
+            "patch": {
+              "patchStrategy": "replace",
+              "mutatedRef": [
+                ["settings", "spec", "jobTemplate", "spec", "container"]
```
Comment (MUzairS15) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1002]:
```suggestion
                ["settings", "spec", "jobTemplate", "spec", "template"]
```

File: server/meshmodel/kubernetes/relationships/hierarchical_inventory.json [hunk 2]
```diff
@@ -988,6 +988,95 @@
...
```
Comment (MUzairS15) [inline server/meshmodel/kubernetes/relationships/hierarchical_inventory.json:1069]:
```suggestion
                ["settings", "spec", "jobTemplate", "spec", "template"]
```

**Pull request merged**